src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,307 @@
+    'plugin/PluginBase',
+    'deepforge-keras/JSONModelMaps',
+    'deepforge-keras/json-model-parser'
+], function (
+    PluginConfig,
+    pluginMetadata,
+    PluginBase,
+    ModelMaps,
+    JSONLayerParser
+) {
+    'use strict';
+
+    pluginMetadata = JSON.parse(pluginMetadata);
+
+
+    /**
+     * Initializes a new instance of ImportKeras.
+     * @class
+     * @augments {PluginBase}
+     * @classdesc This class represents the plugin ImportKeras.
+     * @constructor
+     */
+    var ImportKeras = function () {
+        // Call base class' constructor.
+        PluginBase.call(this);
+        this.pluginMetadata = pluginMetadata;
+    };
+
+    /**
+     * Metadata associated with the plugin. Contains id, name, version, description, icon, configStructure etc.
+     * This is also available at the instance at this.pluginMetadata.
+     * @type {object}
+     */
+    ImportKeras.metadata = pluginMetadata;
+
+    // Prototypical inheritance from PluginBase.
+    ImportKeras.prototype = Object.create(PluginBase.prototype);
+    ImportKeras.prototype.constructor = ImportKeras;
+
+    /**
+     * Main function for the plugin to execute. This will perform the execution.
+     * Notes:
+     * - Always log with the provided logger.[error,warning,info,debug].
+     * - Do NOT put any user interaction logic UI, etc. inside this method.
+     * - callback always has to be called even if error happened.
+     *
+     * @param {function(Error|null, plugin.PluginResult)} callback - the result callback
+     */
+    ImportKeras.prototype.main = function (callback) {
+        let srcJsonHash = this.getCurrentConfig().srcModel;
+        if (!srcJsonHash) {
+            return callback(new Error('Keras Json Not Provided'), this.result);
+        }
+
+        this.archName = this.getCurrentConfig().archName;
+        return this.blobClient.getMetadata(srcJsonHash)
+            .then(metadata => {
+                this.archName = this.archName ? this.archName : metadata.name.replace('.json', '');
+                this.logger.debug(`The new architecture added will be named ${this.archName}`);
+                return this.blobClient.getObjectAsJSON(srcJsonHash);
+            })
+            .then((modelJson) => {
+                this.modelInfo = JSONLayerParser.flatten(modelJson).config;
+                return this.addNewArchitecture()
+                    .then(() => this.addLayers())
+                    .then(() => this.addConnections()
+                        .then(() => this.save('Completed Import Model'))
+                        .then(()=> {
+                            this.result.setSuccess(true);
+                            callback(null, this.result);
+                        })
+                        .catch((err) => {
+                            callback(err, this.result);
+                        }));
+            })
+            .catch((err) => {
+                this.logger.debug(`There is something wrong with Uploaded file.`);

```

Comment:
I would log the actual error (this hides the current error which can make debugging annoying).

Reply:
https://github.com/deepforge-dev/deepforge-keras/pull/119#discussion_r336237940, should address this.